server/src/main/java/com/flipkart/varadhi/web/v1/produce/HeaderValidationHandler.java

```diff
@@ -0,0 +1,53 @@
+package com.flipkart.varadhi.web.v1.produce;
+
+import com.flipkart.varadhi.config.VaradhiOptions;
+import com.flipkart.varadhi.exceptions.ArgumentException;
+import io.vertx.ext.web.RoutingContext;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static com.flipkart.varadhi.MessageConstants.Headers.VARADHI_HEADER_PREFIX;
+
+public class HeaderValidationHandler {
+    private int headerNameSizeMax;
+    private int headerValueSizeMax;
+    private int headersAllowedMax;
+
+    public HeaderValidationHandler(VaradhiOptions varadhiOptions) {
+        this.headerNameSizeMax = varadhiOptions.getHeaderNameSizeMax();
+        this.headerValueSizeMax = varadhiOptions.getHeaderValueSizeMax();
+        this.headersAllowedMax = varadhiOptions.getHeadersAllowedMax();
```

Comment:
are these validations really required? Seems like an unnecessary overhead just to do this.

Reply:
agree on performance part.. but it ensures headers are not used as mechanism to dump arbitrary info.